Sources/GRPCHTTP2Core/Server/GRPCServerStreamHandler.swift

```diff
@@ -99,6 +103,16 @@ extension GRPCServerStreamHandler {
         switch action {
         case .receivedMetadata(let metadata):
           context.fireChannelRead(self.wrapInboundOut(.metadata(metadata)))
+          if let methodDescriptor = metadata.methodDescriptor {
+            self.methodDescriptor.succeed(methodDescriptor)
+          } else {
+            self.methodDescriptor.fail(
+              RPCError(
+                code: .internalError,
+                message: "State machine should have thrown if a method descriptor was not present."
+              )
+            )
+          }
```

Comment:
Can we have these changes (the state machine + channel handler) ones in a separate PR?

Reply:
>Parsing the method descriptor isn't free and we already do it when receiving the metadata, can we just return it with the receivedMetadata case instead?

I didn't want to do this originally because it makes no sense for the action to include the method descriptor when the client receives metadata. But I agree that we want to avoid the slowness of parsing so I'll make this change.

Reply:
https://github.com/grpc/grpc-swift/pull/1891